File: Sources/cpp/src/utils/file_utils.cpp
#include "file_utils.h"

std::vector<std::string> listFiles(const std::string& directory, const std::string& extension) {
    std::vector<std::string> files;
    for (const auto& entry : std::filesystem::directory_iterator(directory)) {
        if (entry.is_regular_file()) {
            auto& path = entry.path();
            if (path.extension() == extension) {
                files.push_back(path.string());
            }
        }
    }
    return files;
}

std::string fileName(const std::string& path) {
    std::filesystem::path fs_path(path);
    return fs_path.stem().string(); 
}

File: Sources/windows/App/Sources/Utils/FileUtils.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

public static class FileUtils
{
    public static List<string> ListFiles(string directory, string extension)
    {
        List<string> files = new List<string>();

        foreach (var file in Directory.EnumerateFiles(directory))
        {
            if (Path.GetExtension(file).Equals(extension, StringComparison.OrdinalIgnoreCase))
            {
                files.Add(file);
            }
        }

        return files;
    }

    public static string FileName(string path)
    {
        return Path.GetFileNameWithoutExtension(path);
    }
}
